src/scitiff/io.py

```diff
@@ -3,78 +3,115 @@
 import json
 import pathlib
 
+import pydantic
 import scipp as sc
 import tifffile as tf
 from scipp.compat.dict import from_dict
 
-
-def _ensure_dimension_order(sizes: dict) -> dict:
-    # Order of the dimensions is according to the HyperStacks tiff format.
-    order = ['c', 't', 'z', 'y', 'x']
-    return {key: sizes[key] for key in order if key in sizes}
```

Comment:
Why did you remove this? The code is still used below but inlined into another function.